flamedisx/xenon/x1t_sr1.py

```diff
@@ -439,40 +443,62 @@ class SR1WallSource(fd.SpatialRateERSource, SR1ERSource):
       # TODO: The parameters here will need to be polished. 
       # They are fitted parameters and give a reasonable result.
       @staticmethod
-      def p_electron(nq, *, w_er_pel_a = -123. , w_er_pel_b = -47.7, 
-                    w_er_pel_c = 68., w_er_pel_e0 = 9.95):
+      def p_electron(nq, *, w_er_pel_c = 0.000174, lam = 0.015):
       
-          """Fraction of ER quanta that become electrons
-          Simplified form from Jelle's thesis
+          """Fraction of ER quanta that become electrons.
+          Exponentially falling yield.
           """
-          # The original model depended on energy, but in flamedisx
-          # it has to be a direct function of nq.
-          e_kev_sortof = nq * 13.7e-3
-          eps = fd.tf_log10(e_kev_sortof / w_er_pel_e0 + 1e-9)
-          qy = (
-              w_er_pel_a * eps ** 2
-              + w_er_pel_b * eps
-              + w_er_pel_c)
-          return fd.safe_p(qy * 13.7e-3)
+          # TODO: Maybe this can be tied to W in a combined fit
+          W = 13.8e-3 #keV
+          eps = (nq*W)
+          qy = w_er_pel_c*tf.exp(-lam*eps)
+          return fd.clip_by_value(qy * W, 1e-8, 1. - 1e-9)
 
       @staticmethod
       def electron_detection_eff(drift_time,
                                  elife,
                                  *,
-                                 w_extraction_eff=0.0169):
+                                 w_extraction_eff=0.89):
           return w_extraction_eff * tf.exp(-drift_time / elife)
-          
+
       @staticmethod
-      def p_electron_fluctuation(nq, w_q2 = 0.0237, w_q3_nq = 123.): 
-        return tf.clip_by_value(
-            w_q2 * (tf.constant(1., dtype=fd.float_type()) - \
-            tf.exp(-nq / w_q3_nq)),
-            tf.constant(1e-4, dtype=fd.float_type()),
-            float('inf'))
-      # It is preferred to have higher energy spectrum for the wall
-      energies = tf.cast(tf.linspace(0., 25. , 1000),
-                       dtype=fd.float_type())
+      def electron_loss(electrons_produced,
+                        *,
+                        l_max=0.084,
+                        alpha=0.0001,
+                        l_min=0.002):
+          return l_max*(tf.exp(-alpha*electrons_produced))+l_min
```

Comment:
As discussed, since `electron_loss` is a value between [0,1], there should be a `safe_p` wrapping this, or a `tf.clip_by_value` if a typical value is constrained to an even smaller range than [0,1].

Reply:
solved